webui/react/src/components/FilterForm/components/ConjunctionContainer.test.tsx

```diff
@@ -0,0 +1,57 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { SelectValue } from 'hew/Select';
+import UIProvider, { DefaultTheme } from 'hew/Theme';
+
+import ConjunctionContainer from 'components/FilterForm/components/ConjunctionContainer';
+
+import { Conjunction } from './type';
+
+const setup = ({
+  index = 0,
+  conjunction,
+  onClick = vi.fn(),
+}: {
+  index?: number;
+  conjunction: Conjunction;
+  onClick?: (value: SelectValue) => void;
+}) => {
+  const user = userEvent.setup();
+
+  render(
+    <UIProvider theme={DefaultTheme.Light}>
+      <ConjunctionContainer conjunction={conjunction} index={index} onClick={onClick} />
+    </UIProvider>,
+  );
+
+  return { user };
+};
+
+describe('ConjunctionContainer', () => {
+  it('should show Where text when index is 0', () => {
+    setup({ conjunction: Conjunction.Or });
+
+    expect(screen.getByText('Where')).toHaveAttribute('data-test', 'where');
+    expect(screen.queryByRole('combobox')).not.toBeInTheDocument();
+  });
+
+  it('should show Select when index is 1', async () => {
+    const { user } = setup({ conjunction: Conjunction.Or, index: 1 });
+
+    const conjunctionSelect = screen.getByRole('combobox');
+    expect(screen.queryByText('Where')).not.toBeInTheDocument();
+    expect(conjunctionSelect).toBeInTheDocument();
+
+    expect(screen.queryByRole('listbox')).not.toBeInTheDocument();
+    await user.click(conjunctionSelect);
+    expect(await screen.findByRole('listbox')).toBeInTheDocument();
```

Comment:
Would prefer not just testing presence of these elements but also that selecting a new conjunction calls `onClick` with the expected value

Reply:
fixed it